Source/EmbeddedBoundary/WarpXInitEB.cpp

```diff
@@ -241,9 +236,72 @@ WarpX::ScaleAreas() {
                 full_area = cell_size[0]*cell_size[1];
             }
             auto const &face_areas_dim = m_face_areas[maxLevel()][idim]->array(mfi);
-            amrex::LoopOnCpu(amrex::convert(box, amrex::Box(face_areas_dim).ixType()),
-                             [=](int i, int j, int k) {
+            auto const &mod_areas_dim = m_area_mod[maxLevel()][idim]->array(mfi);
+
+            amrex::ParallelFor(box, [=](int i, int j, int k) {
                                 face_areas_dim(i, j, k) *= full_area;
+                                mod_areas_dim(i, j, k) = face_areas_dim(i, j, k);
+            });
+        }
+    }
+#endif
+}
+
+//0 for unst, 1 for stable and available, 2 for stable, available and intruded
+
+/**
+ * \brief Initialize information for cell extensions.
+ *        The flags convention for m_flag_info_face is as follows
+ *          - 0 for unstable cells
+ *          - 1 for stable cells which have not been intruded
+ *          - 2 for stable cells which have been intruded
+ *        Here we cannot know if a cell is intruded or not so we initialize all stable cells with 1
+ */
+void
+WarpX::MarkCells(){
+#ifdef AMREX_USE_EB
+    auto const &cell_size = CellSize(maxLevel());
+
+    for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+        for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+            amrex::Box const &box = mfi.tilebox(m_face_areas[maxLevel()][idim]->ixType().toIntVect());
+
+            auto const &S = m_face_areas[maxLevel()][idim]->array(mfi);
+            auto const &flag_info_face = m_flag_info_face[maxLevel()][idim]->array(mfi);
+            auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+            const auto &lx = m_edge_lengths[maxLevel()][0]->array(mfi);
+            const auto &ly = m_edge_lengths[maxLevel()][1]->array(mfi);
+            const auto &lz = m_edge_lengths[maxLevel()][2]->array(mfi);
+            amrex::Real dx = cell_size[0];
+            amrex::Real dy = cell_size[1];
+            amrex::Real dz = cell_size[2];
+
+            amrex::ParallelFor(box, [=](int i, int j, int k) {
```

Comment:
Shall we add this, too? Cc: @WeiqunZhang 
```suggestion
            amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE (int i, int j, int k) {
```